internal/v2/fakes2av2/fakes2av2.go

```diff
@@ -98,15 +99,19 @@ func offloadPrivateKeyOperation(req *s2av2pb.OffloadPrivateKeyOperationReq, host
 			s := fmt.Sprintf("Invalid hostname tied to SessionReq: %s", hostname)
 			return nil, errors.New(s)
 		}
-
-		// TODO(rmehta19): Parse req.GetSignatureAlgorithm() to find Hash function that
-		// should be used here. Currently hardcoded to use SHA256.
-
-		// Sign the SHA256 hash with the private key.
-		var opts crypto.Hash = crypto.SHA256
-		signedBytes, err := root.PrivateKey.(crypto.Signer).Sign(rand.Reader, req.GetInBytes(), opts)
-		if err != nil {
-			return nil, err
+		var signedBytes []byte
+		if req.GetSignatureAlgorithm() == s2av2pb.SignatureAlgorithm_S2A_SSL_SIGN_RSA_PKCS1_SHA256 {
+			var opts crypto.Hash = crypto.SHA256
+			signedBytes, err = root.PrivateKey.(crypto.Signer).Sign(rand.Reader, req.GetInBytes(), opts)
+			if err != nil {
+				return nil, err
+			}
+		} else if req.GetSignatureAlgorithm() == s2av2pb.SignatureAlgorithm_S2A_SSL_SIGN_RSA_PSS_RSAE_SHA256 {
+			var opts = rsa.PSSOptions { -1, crypto.SHA256 }
```

Comment:
For my own understanding, why is the salt length equal to -1?

Reply:
-1 is a constant: https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/crypto/rsa/pss.go;l=231;bpv=0, perhaps I can use PSSSaltLengthEqualsHash, the named variable to make this more readable.

This is what the go crypto/tls calls the top - level Sign function(defined in remote_signer.go) with. For now, I've hardcoded it here, since this is the case for the go echo client and server binaries, during their client and server handshakes. However, perhaps there is a way we can retrieve this info(the opts variable) from the Client / Server Hello and pass it through to S2A through an additional proto field in OffloadPrivateKeyReq